### ISSUE TITLE
Triplet Loss Slice Index Error

### ISSUE THREAD

**Issue by nailcankara** [keras-team/keras#14263]
```
Layer (type)                    Output Shape         Param #     Connected to                     
==================================================================================================
anchor (InputLayer)             (None, 160, 160, 3)  0                                            
__________________________________________________________________________________________________
anchorPositive (InputLayer)     (None, 160, 160, 3)  0                                            
__________________________________________________________________________________________________
anchorNegative (InputLayer)     (None, 160, 160, 3)  0                                            
__________________________________________________________________________________________________
inception_resnet_v1 (Model)     (None, 128)          22808144    anchor[0][0]                     
                                                                 anchorPositive[0][0]             
                                                                 anchorNegative[0][0]             
==================================================================================================
Total params: 22,808,144
Trainable params: 0
Non-trainable params: 22,808,144
```

My model is this and my loss is

```
def triplet_loss(y_true, y_pred):
    alpha = 0.2

    anchor, positive, negative = y_pred[0], y_pred[1], y_pred[2],
    pos_dist = tf.reduce_sum(tf.square(tf.subtract(anchor, positive)), axis=-1)
    neg_dist = tf.reduce_sum(tf.square(tf.subtract(anchor, negative)), axis=-1)

    basic_loss = tf.add(tf.subtract(pos_dist, neg_dist), alpha)
    loss = tf.reduce_sum(tf.maximum(basic_loss, 0.0))
    return loss 
```

this. But in the loss function , my **y_pred** variable's shape is (1,128). Shouldn't be (3,128)? Always I am getting slice index error.

**Comment by sushreebarsa** [comment]
@nailcankara 
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thanks!
